digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala

```diff
@@ -382,29 +402,69 @@ class SpeedLimitsTierekisteriImporter extends LinearAssetTierekisteriImporterOpe
     }
   }
 
-  private def getSpeedLimitValue(trAsset: TierekisteriAssetData) = {
+  private def getSpeedLimitValue(assetInfo: TrAssetInfo, linkType:  Option[LinkType]): Option[Int] = {
+    def getSpeedLimitValueByLinkType(linkType:  Option[LinkType]): Option[Int] = {
+    linkType.getOrElse("") match {
+      case Motorway =>
+        Some(defaultMotorwaySpeedLimit)
+      case MultipleCarriageway | SingleCarriageway | Freeway =>
+        Some(defaultCarriageOrFreewaySpeedLimit)
+      case _ =>
+        None
+    }
+  }
+    val trAsset = assetInfo.trAsset.asInstanceOf[TierekisteriAssetData]
+    print("assetType" -> trAsset.asInstanceOf[TierekisteriAssetData].assetType.trafficSignType)
     trAsset.assetType.trafficSignType match {
       case TrafficSignType.SpeedLimit  =>
+        println("")
         toInt(trAsset.assetValue)
       case TrafficSignType.EndSpeedLimit =>
+        println("")
         Some(defaultSpeedLimit)
       case TrafficSignType.SpeedLimitZone  =>
+        println("")
         toInt(trAsset.assetValue)
       case TrafficSignType.EndSpeedLimitZone =>
+        println("")
         Some(defaultSpeedLimit)
       case TrafficSignType.UrbanArea =>
+        println("")
         Some(urbanAreaSpeedLimit)
       case TrafficSignType.EndUrbanArea =>
+        println("")
         Some(defaultSpeedLimit)
+      case TrafficSignType.TelematicSpeedLimit =>
+        getSpeedLimitValueByLinkType(linkType) match {
+          case None => getSpeedLimitValueByLinkType(assetInfo.linkType)
+          case value => value
```

Comment:
Can be modified to:
**getSpeedLimitValueByLinkType(linkType) match {
          case value => value
          case _ => getSpeedLimitValueByLinkType(assetInfo.linkType)**